assets/controllers/disaffiliate_user_controller.js

```diff
@@ -0,0 +1,48 @@
+import {Controller} from '@hotwired/stimulus';
+import {axiosInstance} from '../js/helpers/axios';
+import Swal from "sweetalert2";
+
+export default class extends Controller {
+  static targets = ['icon']
+  static values = {
+    url: String,
+    confirmMessage: String,
+    confirmButtonText: String,
+    cancelButtonText: String,
+  };
+  swalOptions = {
+    text: this.confirmMessageValue,
+    icon: 'warning',
+    showCancelButton: true,
+    confirmButtonText: this.confirmButtonTextValue,
+    cancelButtonText: this.cancelButtonTextValue,
+    reverseButtons: true,
+    customClass: {
+      cancelButton: 'btn btn-outline-black me-1',
+      confirmButton: 'btn btn-red text-white ms-1'
+    },
+    buttonsStyling: false,
+    showLoaderOnConfirm: true,
+  }
+
+  disaffiliate(event) {
+    const button = event.currentTarget;
+
+    Swal.fire({
+      ...this.swalOptions,
+      preConfirm: () => {
+        return axiosInstance.get(this.urlValue)
+          .then(function () {
+            this.replaceClasses(this.iconTarget, 'text-primary', 'text-grey');
+            this.replaceClasses(button, 'btn-red', 'btn-grey');
+            button.removeAttribute('data-action');
+          }.bind(this))
```

Comment:
pourquoi ce bind ?

Reply:
Pour avoir accès à this.replaceClasses()